### PR TITLE
[TUTORIAL] Revise Naming tutorial

### PR DESCRIPTION
**Pull request by Ishitori**
## Description ##
Update the text of Naming tutorials to make it more readable. Plus adding conclusion and next step sections.

### CODE REVIEW

File: docs/tutorials/gluon/naming.md [hunk 1]
```diff
@@ -137,16 +116,11 @@ print(model1.dense1.prefix)
...
-    model1_mydense_
-
-
-**It is recommended that you manually specify a prefix for the top level Block, i.e. `model = Model(prefix='mymodel_')`, to avoid potential confusions in naming.**
+Note that `model1.dense0` is still named as `dense0_` instead of `dense2_`, following `Dense` layers in previously created `model0`. This is because each instance of model's name scope is independent of each other.
 
-The same principle also applies to container blocks like Sequential. `name_scope` can be used inside `__init__` as well as out side of `__init__`:
+**It is recommended that you manually specify a prefix for the top level `Block`, i.e. `model = Model(prefix='mymodel_')`, to avoid potential confusion in naming.**
 
+The same principle  applies to container blocks like [`Sequential`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/gluon.html#mxnet.gluon.nn.Sequential). `name_scope` can be used inside as well as outside of `__init__`:
```
Comment (thomelane) [inline docs/tutorials/gluon/naming.md]: The same principle  applies -> The same principle applies

Comment (thomelane) [inline docs/tutorials/gluon/naming.md]: Can tell whether you're saying this is recommended or not? Using name scope with Sequential?

File: docs/tutorials/gluon/naming.md [hunk 2]
```diff
@@ -174,41 +142,17 @@ with net.name_scope():
...
-      Parameter model1_dense0_weight (shape=(20, 0), dtype=<type 'numpy.float32'>)
-      Parameter model1_dense0_bias (shape=(20,), dtype=<type 'numpy.float32'>)
-      Parameter model1_dense1_weight (shape=(20, 0), dtype=<type 'numpy.float32'>)
-      Parameter model1_dense1_bias (shape=(20,), dtype=<type 'numpy.float32'>)
-      Parameter model1_mydense_weight (shape=(20, 0), dtype=<type 'numpy.float32'>)
-      Parameter model1_mydense_bias (shape=(20,), dtype=<type 'numpy.float32'>)
-    )
-
-
 As a result, if you try to save parameters from model0 and load it with model1, you'll get an error due to unmatching names:
```
Comment (thomelane) [inline docs/tutorials/gluon/naming.md]: "load it with model1"? Might want to rephrase this.

File: docs/tutorials/gluon/naming.md [hunk 3]
```diff
@@ -217,45 +161,31 @@ except Exception as e:
...
-Sometimes you may want to load a pretrained model, and replace certain Blocks in it for fine-tuning.
-
-For example, the alexnet in model zoo has 1000 output dimensions, but maybe you only have 100 classes in your application.
+## Replacing Blocks in networks and fine-tuning
 
-To see how to do this, we first load a pretrained AlexNet.
-
-- In Gluon model zoo, all image classification models follow the format where the feature extraction layers are named `features` while the output layer is named `output`.
-- Note that the output layer is a dense block with 1000 dimension outputs.
+Sometimes you may want to load a pretrained model, and replace certain Blocks in it for fine-tuning. For example, the [`AlexNet`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/model_zoo.html#vision) model in the model zoo has 1000 output dimensions, but maybe you have only 100 classes in your application. Let's see how to change the number of output dimensions from 1000 to 100.
```
Comment (thomelane) [inline docs/tutorials/gluon/naming.md]: Would mention that you can just set `classes` argument for this case. Guess you're doing like this for demonstration purposes.

**Comment by thomelane** [comment]
Great, thanks for updates. LGTM.

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Pull request merged**